src/Query.jsx

```diff
@@ -0,0 +1,97 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { ArrowBackUp, Database } from 'tabler-icons-react';
+import { Button, Paper, Space } from '@mantine/core';
+/* eslint-disable import/no-named-default */
+import { default as Input } from './HighlightInput';
+// import { default as Input } from './PrismInput';
+import DataTable from './DataTable';
+
+const usePrevious = (value) => {
+  const ref = React.useRef();
+  React.useEffect(() => {
+    ref.current = value;
+  });
+  return ref.current;
+};
+
+export default function Query({ execute, initialQuery }) {
+  const [queryValue, setQueryValue] = React.useState(initialQuery || '');
+  const [queryResult, setQueryResult] = React.useState();
+
+  const buttonRef = React.useRef();
+  const editorRef = React.useRef();
+
+  const handleExecute = () => {
+    setQueryResult(execute(queryValue));
+  };
+  const handleReset = () => {
+    setQueryResult();
+  };
+
+  const onKeyDown = (event) => {
+    if (event.key === 'Enter' && event.shiftKey) {
+      event.stopPropagation();
+      handleExecute();
+    }
+  };
+
+  const previousQueryResult = usePrevious(queryResult);
+  React.useEffect(() => {
+    if (!queryResult && previousQueryResult) {
+      editorRef.current.focus();
+    } else if (queryResult && !previousQueryResult) {
+      buttonRef.current.focus();
+    }
+  }, [queryResult, previousQueryResult]);
+
+  return (
+    <Paper p="sm" radius="sm" shadow="md" withBorder>
+      <Input
+        disabled={Boolean(queryResult)}
+        onChange={setQueryValue}
+        onKeyDown={onKeyDown}
+        ref={editorRef}
+        value={queryValue}
+      />
```

Comment:
i think (haven't tried it) that if you used `this.props.children` to get the input, you could remove that renamed `Input` import from L6 and define `Input` before the render like this:

```jsx
# https://reactjs.org/docs/react-api.html#reactchildrenonly
# Verifies that children has only one child (a React element) and returns it. Otherwise this method throws an error.
const Input = React.Children.only(children); 
```

i'd expect the `Query` component to then error if it isn't used with a react component as its child, like this:
```
<Query>
    <!-- or whatever input someone drops in here -->
    <Textarea />
</Query>
```